### ISSUE TITLE
mdtool: Could not locate executable.

### ISSUE THREAD

**Issue by jzeferino** [Redth/Cake.Xamarin#10]
On OS X I got this error:

> mdtool: Could not locate executable.

Running:
 			`MDToolSetup.Pack(binaryPath, artifactsDir);`

Full stack trace:

> at Cake.Core.Tooling.Tool`1[TSettings].RunProcess (TSettings settings, Cake.Core.IO.ProcessArgumentBuilder arguments, Cake.Core.IO.ProcessSettings processSettings) [0x0009a] in <47bd6426c6a64c7ea6de1ba15c2457b7>:0 
>   at Cake.Core.Tooling.Tool`1[TSettings].Run (TSettings settings, Cake.Core.IO.ProcessArgumentBuilder arguments, Cake.Core.IO.ProcessSettings processSettings, System.Action`1[T] postAction) [0x00026] in <47bd6426c6a64c7ea6de1ba15c2457b7>:0 
>   at Cake.Core.Tooling.Tool`1[TSettings].Run (TSettings settings, Cake.Core.IO.ProcessArgumentBuilder arguments) [0x00001] in <47bd6426c6a64c7ea6de1ba15c2457b7>:0 
>   at Cake.Xamarin.MDToolSetupRunner.Pack (Cake.Core.IO.FilePath addinFile, Cake.Core.IO.DirectoryPath outputDirectory) [0x0004d] in <279329ce540b49f2bdca05af04b9e327>:0 
>   at CakeBuildScriptImpl+<Execute>c__AnonStorey0.<>m__5 () [0x00021] in <df1a31afb64f419887539b5a5cdfdb28>:0

**Comment by agc93** [comment]
@jzeferino Is `mdtool` in your PATH? Cake, by default, will only resolve from PATH since it doesn't know your Xamarin Studio/VS for Mac install path.

**Comment by jzeferino** [comment]
It ins't in my PATH, but since this is Cake.Xamarin, it would be nice to check if the mdtool was in the common places (in xamarin studio bin for example).

So, since the cake could run in (eg: windows, linux, osx) I will have to Register the mdtool.exe when cake runs in the different OS types. And in the windows case, the mdtool.exe ins't shipped with visual studio (with xamarin) that I know.

**Comment by agc93** [comment]
@jzeferino Yeah, so the paths change based not just on platform but installation location, IDE version and a bunch of other variables (so even 'Xamarin Studio bin' is not  so simple), so you may need to provide it with the path to `mdtool`/`mdtool.exe` if it's not in your PATH.

Unless @Redth know more about resolving `mdtool` paths than me, perhaps..

**Comment by jzeferino** [comment]
For now I have this (just to Pack locally):

```c#
Task("Pack")
	.IsDependentOn("Build")
    .WithCriteria(() => isLocalBuild)
    .WithCriteria(() => HasMdTool())
	.Does(() => 
	{
		MDToolSetup.Pack(binaryPath, artifactsDir);
	});

private bool HasMdTool()
{
	var mdToolPath = @"/Applications/Xamarin Studio.app/Contents/Resources/lib/monodevelop/bin/mdtool.exe";

	if (FileExists(mdToolPath)) {
		Information("mdtool exists");

		Context.Tools.RegisterFile(mdToolPath);
		return true;
	}
	return false;
}
```

Would like to hear @Redth on this.

**Comment by Redth** [comment]
I don't believe I make any current assumptions as to where mdtool exists.  Definitely a good idea to at least try for some well known paths.
